### PR TITLE
Improve OneHot encoder

### PR DESCRIPTION
**Pull request by ganesh-krishnan**
## What
- Fix names for `OneHot` encoded columns. The column names were not being set properly
- Add `drop_first` option to drop first level
- Add option to specify `percent_occurrences` to prune rare levels
- Add documentation for `OneHot`
- Add additional tests for `OneHot`

## Why
Because why not?

### CODE REVIEW

File: lore/encoders.py [hunk 1]
```diff
@@ -575,18 +575,31 @@ def cardinality(self):
...
     """
-    Performs one hot encoding
+    Performs one hot encoding. One hot encoding is used to encode categorical variables into a series of indicator variables.
     """
-    def __init__(self, column, name=None, minimum_occurrences=None, compressed=False, **kwargs):
-        if compressed is True and minimum_occurrences is None:
-            raise ValueError('minimum_occurrences must be specified when compressed is True')
-        elif compressed is False and minimum_occurrences is not None:
-            logger.warning('minimum_occurrences has no effect when compressed is False')
+    def __init__(self, column, name=None, minimum_occurrences=None, percent_occurrences=None,
```
Comment (sarchak) [inline lore/encoders.py]: If I understand this correctly what we are doing is if the categorical variable count or % wise is less than some factor x we just set them to 0? Can you point me to some more information about the intuition behind this?

Comment (ganesh-krishnan) [reply lore/encoders.py]: In one hot encoding, each of the variables represents the additional effect of that categorical level. If you have "rare" levels, you are essentially saying that they are not contributing any "additional" effect. That's one way to look at it. Alternatively, you could say that the rare levels are being pooled into one common level.

File: lore/encoders.py [hunk 2]
```diff
@@ -575,18 +575,31 @@ def cardinality(self):
...
+            logger.warning('minimum_occurrences and percent_occurences have no effect when compressed is False')
+        self.percent_occurrences = percent_occurrences
         self.minimum_occurrences = minimum_occurrences
+        self.drop_first = drop_first
         self.compressed = compressed
         super(OneHot, self).__init__(column, name, **kwargs)
 
     def fit(self, data):
+        if self.percent_occurrences is not None:
+            self.minimum_occurrences = self.percent_occurrences*len(self.series(data))
```
Comment (montanalow) [inline lore/encoders.py]: Stylistically, mathematical operators should have whitespace around them, I need to add a linter, but then I'd have to abide by it too.

File: lore/encoders.py [hunk 3]
```diff
@@ -595,6 +608,10 @@ def fit(self, data):
         else:
             self.categories = list(ids.id.unique())
 
+        if len(self.categories) <= 1:
```
Comment (montanalow) [inline lore/encoders.py]: Nice sanity check. @abhayspawar has a nice lib that does this sort of check for all encoders. Would be nice to get it in :)

**Pull request merged**